### DISCUSSION TITLE
Setup of HA

### DISCUSSION THREAD

**Discussion by wanavt** [macxq/foxess-ha#258] [Q&A] [answered]
Hi all, apologies if some of the below questions are simple - I am still quite new to HA. I am trying to setup my solar panel system into HA. I have two inverters (one Hybrid and one PV) connected to various panels with 12.9kWh of FoxEss batteries

When looking at the entities I see various foxess entities but they all have the following error: This entity is no longer being provided by the foxess integration. If the entity is no longer in use, delete it in settings.

My system was installed in Jan24 but this is the first time I am integrating it into HA. I tried using the SN in the for the ID but that didnt seemed to work either. I also am using the same API key for both inverters - is this correct? I cannot seem to create a 2nd API key

sensor:
#Install FoxESS
  - platform: foxess
    name: HybridInverter
    deviceID: 6c48085d-b81e-4a55-b022-30e3e5f2xxxx
    deviceSN: 60BH5020377N126
    apiKey: xxx
  - platform: foxess
    name: PVInverter
    deviceID: ab14c2ab-6412-40b6-8975-0cb09dd5xxxx
    deviceSN: 601F50202BXF154 
    apiKey: xxx

Many thanks for your help!

**Comment by FozzieUK** [comment, marked as answer]
The setup looks ok, if you have both inverters under the same account there will only be the one apiKey - when you are in the API management section of the Fox website you'll see the number of remaining api calls, which is 1,440 per day - if it says 2,880 that confirms they both share the same apiKey.

The apiKey is a oneshot only thing, so you must make sure that you create a new one - copy it immediately and then paste it into the apiKey section of your home assistant platform: foxess, don't request another one as that will invalidate the first.

If your HA has created the entities, it could be you need to generate a new apiKey and use that.

Once you've done that, restart your HA and when it has started leave it for 5 minutes, then click on Settings, System, Logs and towards the bottom click on the 'LOAD FULL LOGS' button - it will show all error logs and if you could copy and paste any logs that belong to 'foxess' that would help me work out why if there are still problems.

Just one last caveat, the OpenAPI is having performance problems at the moment - it works ok in the morning but during the afternoon and early evening it is very slow and that will cause some errors to be generated and it may occasionally see the sensors go 'unknonw' but this integration will keep retrying and as soon as it gets a response it will update the sensors.

Finally make sure you have updated to the latest version of the integration v0.41

**Reply by wanavt** [reply]
Thanks for the reply Fozzie - appreciate it, Correct I can see 2880 calls per day and also I am on version v0.41

I have created a new API key and added that to my config. Restarting HA and waited 5mins. Log extracts as below - please let me know if you need anything else

2024-06-19 17:25:46.196 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration foxess which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

2024-06-19 17:26:00.359 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform foxess is taking over 10 seconds.
2024-06-19 17:26:00.364 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform foxess is taking over 10 seconds.

2024-06-19 17:26:48.978 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: {('foxess', '547478280112'): 467600.927348011, ('foxess', '547478279632'): 467600.932552661, ('motion_blinds', '496371663d2844e1e83b2bd7fd629490'): 467615.4421715}
2024-06-19 17:26:50.363 ERROR (MainThread) [homeassistant.components.sensor] Setup of platform foxess is taking longer than 60 seconds. Startup will proceed without waiting any longer.
2024-06-19 17:26:50.367 ERROR (MainThread) [homeassistant.components.sensor] Setup of platform foxess is taking longer than 60 seconds. Startup will proceed without waiting any longer.

**Reply by FozzieUK** [reply]
Ok that does look reasonable, the foxess integration is being held up by the OpenAPI which means it’s definitely making the calls - and that corresponds with what i’m seeing when I look at the response time.

Can you leave it approx another 30 minutes, then check to see if you have any data particularly the sums such as sensor.{your name}_energy_generated as they will show data the instant they receive any.

If there is no data, can you repeat the logs and you should see more information if there is an error

**Reply by FozzieUK** [reply]
The OpenAPI is having a real go slow right now, my test system is not connecting either - it might take a bit longer until it settles down.

![IMG_1619](https://github.com/macxq/foxess-ha/assets/113460294/3e0bbfc5-17ee-438c-913c-6a04532065d0)

**Reply by wanavt** [reply]
Thanks for the responses Fozzie, I left it overnight and I see data! Looks like I have three sets of names, not sure if it was an error when I tried setting it up - can I delete the 'foxess' ones (which have an error attached to them)? Is there an easier way instead of manually clicking each one > settings > delete

Thanks again for all the help!

Data for sensor.hybridinverter_energy_generated
![image](https://github.com/macxq/foxess-ha/assets/102619124/849211ff-4230-4706-a411-43caee31c752)

Data for sensor.pvinverter_energy_generated
![image](https://github.com/macxq/foxess-ha/assets/102619124/f9257d1d-250f-4cbb-9bc8-ed7341652f8e)

Three sets (actually 4)
![image](https://github.com/macxq/foxess-ha/assets/102619124/283d6d34-05aa-4b8d-8787-a7fd52f46ae9)

**Reply by FozzieUK** [reply]
Ok that’s great, all looks good.

It’s a shame the OpenAPI is playing up and you’ll see the power sensors go ‘unknown’ occasionally during the day, but it’ll re-connect automatically as soon as it’s back - Fox ESS know it needs resources to resolve, and it’s also affecting their corporate clients so hopefully not much longer.

The foxess entities are the default names so probably happened at some point when you first set it up before adding an inverter name, you can just delete them. 👍
